### PR TITLE
Two mkosi CI improvements

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: .github/workflows/mkosi.yml [hunk 1]
```diff
@@ -188,7 +188,7 @@ jobs:
 
     - name: Archive failed test journals
       uses: actions/upload-artifact@v4
-      if: failure()
+      if: failure() && github.repository == 'systemd/systemd' || github.repository == 'systemd/systemd-stable'
```
Comment (mrc0mmand) [inline .github/workflows/mkosi.yml]: Shouldn't the OR expression be in parentheses, so this evaluates properly?

File: .github/workflows/mkosi.yml [hunk 2]
```diff
@@ -197,7 +197,7 @@ jobs:
 
     - name: Archive packages
       uses: actions/upload-artifact@v4
-      if: success() || failure()
+      if: (success() || failure()) && github.repository == 'systemd/systemd' || github.repository == 'systemd/systemd-stable'
```
Comment (mrc0mmand) [inline .github/workflows/mkosi.yml]: ditto

Comment (mrc0mmand) [reply .github/workflows/mkosi.yml]: Also, maybe you can just use `always()` instead of `success() || failure()`?

Comment (daandemeyer) [reply .github/workflows/mkosi.yml]: @mrc0mmand `always()` means when cancelled as well, and we don't want that.

**Review by mrc0mmand — APPROVED** [review]

**Pull request merged**